### ISSUE TITLE
Ambiguous use of init(forPlot:withBlock) with CPTFunctionDataSource in Xcode 9.3

### ISSUE THREAD

**Issue by TheTrueTom** [core-plot/core-plot#372]
Since Xcode 9.3 I have this error popping out when I try to compile:

`error: ambiguous use of 'init(forPlot:withBlock:)'`

The piece of code triggering the error is just the following:

`fitFunctionDataSource = CPTFunctionDataSource.init(forPlot: fitPlot, withBlock: regression)`

with `regression` being defined as a `((Double) -> Double)` and `fitPlot` as a `CPTPlot`

I have tried casting anyway I can but to no avail.

The log gives me the following:

```
/Users/***/Documents/***/***/Programme/***/***/MeasurementViewController+Plot.swift:320:45: error: ambiguous use of 'init(forPlot:withBlock:)'
                    fitFunctionDataSource = CPTFunctionDataSource.init(forPlot: fitPlot, withBlock: regression)
                                            ^
CorePlot.CPTFunctionDataSource:25:12: note: found this candidate
    public init(forPlot plot: CPTPlot, withBlock block: @escaping CorePlot.CPTDataSourceBlock)
           ^
CorePlot.CPTFunctionDataSource:17:21: note: found this candidate
    open class func `init`(forPlot plot: CPTPlot, withBlock block: @escaping CorePlot.CPTDataSourceBlock) -> Self
```

If you have any idea on how to fix this I would be grateful.

**Comment by eskroch** [comment]
What version of Core Plot are you using? Try the `release-2.3` branch if you're not already using that.

**Comment by TheTrueTom** [comment]
I'm using the latest from the `release-2.3` branch. The problem started appearing after the update to `Xcode 9.3`

**Comment by eskroch** [comment]
Fixed on the `release-2.3` branch.
